cpp/include/cudf/io/functions.hpp

```diff
@@ -237,6 +238,87 @@ struct read_csv_args {
 table_with_metadata read_csv(read_csv_args const& args,
                              rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+
+/**
+ * @brief Settings to use for `write_csv()`
+ */
+struct write_csv_args: detail::csv::writer_options {
+  write_csv_args(sink_info const& snk,
+                 table_view const& table,
+                 std::string const& na,
+                 bool include_header,
+                 int rows_per_chunk,
+                 std::string line_term = std::string{"\n"},
+                 char delim = ',',
+                 std::string true_v = std::string{"true"},
+                 std::string false_v = std::string{"false"},
+                 table_metadata const* metadata = nullptr)
```

Comment:
The `metadata` parameter defaults to `nullptr`, but using the default value causes it to fail here https://github.com/rapidsai/cudf/blob/682104b57da8af25f9e0e08e623b9d308205f592/cpp/src/io/csv/writer_impl.cu#L389

Reply:
Fixed. Looks like `metadata == nullptr` is a possibility (from the code in the other writers) so just added a test guard and removed `CUDF_EXPECTS`.